### PR TITLE
Issue #9142: Migrate to Truth in AutomaticBeanTest

### PR DESCRIPTION
**Pull request by shashwatj07**
Issue #9142

### CODE REVIEW

**Comment by romani** [comment]
Github, rebase

File: config/sevntu_suppressions.xml [hunk 1]
```diff
@@ -44,5 +44,6 @@
   <suppress checks="RequireFailForTryCatchInJunit" files="MutableExceptionCheckTest.java"/>
   <suppress checks="RequireFailForTryCatchInJunit" files="CheckstyleAntTaskTest.java"/>
   <suppress checks="RequireFailForTryCatchInJunit" files="AbstractViolationReporterTest.java"/>
+  <suppress checks="RequireFailForTryCatchInJunit" files="AutomaticBeanTest.java"/>
```
Comment (romani) [inline config/sevntu_suppressions.xml:47]: http://sevntu-checkstyle.github.io/sevntu.checkstyle/#1.39.0 this Check suppose to support Truth, if still a problem please create ticket against this Check.
and place link to here as reason `<!-- until ..... --> `  to know when to remove this suppression.

**Review by romani — APPROVED** [review]
approved with comment:

File: config/sevntu_suppressions.xml (hunk 1 above)
Comment (shashwatj07) [reply config/sevntu_suppressions.xml:47]: Fix is merged.
Please address this https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues/847#issuecomment-856336514

Comment (timurt) [reply config/sevntu_suppressions.xml:47]: @shashwatj07 please check and and try to eliminate suppressions if `RequireFailForTryCatchInJunit` supports truth fail

Comment (shashwatj07) [reply config/sevntu_suppressions.xml]: They can't be eliminated yet https://cloud.drone.io/checkstyle/checkstyle/3902/1/2
sevntu-checkstyle hasn't been released yet

**Comment by romani** [comment]
@shashwatj07 , please rebase and resolve conflict.

File: config/sevntu_suppressions.xml (hunk 1 above)
Comment (romani) [reply config/sevntu_suppressions.xml:48]: just released 1.40.0 just for you, please bump version in this commit.

Comment (romani) [reply config/sevntu_suppressions.xml:48]: done https://github.com/checkstyle/checkstyle/commit/0d9b3afe600b247eaccf758a0c6ae2bc8409a268

Comment (shashwatj07) [reply config/sevntu_suppressions.xml]: removed

**Comment by shashwatj07** [comment]
> @shashwatj07 , please rebase and resolve conflict.

@romani done

File: src/test/java/com/puppycrawl/tools/checkstyle/api/AutomaticBeanTest.java [hunk 2]
```diff
@@ -193,14 +225,28 @@ public void testBeanConverters() throws Exception {
...
-        assertEquals(SeverityLevel.ERROR, bean.severityLevel, "invalid result");
-        assertEquals(Scope.PUBLIC, bean.scope, "invalid result");
-        assertEquals(new URI("http://github.com"), bean.uri, "invalid result");
-        assertArrayEquals(new AccessModifierOption[] {AccessModifierOption.PUBLIC,
-            AccessModifierOption.PRIVATE}, bean.accessModifiers,
-                "invalid result");
+        final String message = "invalid result";
+        assertWithMessage(message)
+                .that(bean.strings)
+                .isEqualTo(new String[] {"a", "b", "c"});
```
Comment (pbludov) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/AutomaticBeanTest.java:231]: Please avoid using `isEqualTo` for arrays and collections. Suggested change
```
        assertWithMessage(message)
                .that(bean.strings).asList()
                .containsExactly("a", "b", "c");
``` 
It produces more readable violation message.

File: src/test/java/com/puppycrawl/tools/checkstyle/api/AutomaticBeanTest.java [hunk 3]
```diff
@@ -193,14 +225,28 @@ public void testBeanConverters() throws Exception {
...
+                .isEqualTo(SeverityLevel.ERROR);
+        assertWithMessage(message)
+                .that(bean.scope)
+                .isEqualTo(Scope.PUBLIC);
+        assertWithMessage(message)
+                .that(bean.uri)
+                .isEqualTo(new URI("http://github.com"));
+        assertWithMessage(message)
+                .that(bean.accessModifiers)
+                .isEqualTo(new AccessModifierOption[] {
```
Comment (pbludov) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/AutomaticBeanTest.java:246]: same here

**Review by pbludov — CHANGES_REQUESTED** [review]
Two minor notes:

File: src/test/java/com/puppycrawl/tools/checkstyle/api/AutomaticBeanTest.java (hunk 2 above)
Comment (shashwatj07) [reply src/test/java/com/puppycrawl/tools/checkstyle/api/AutomaticBeanTest.java]: done

File: src/test/java/com/puppycrawl/tools/checkstyle/api/AutomaticBeanTest.java (hunk 3 above)
Comment (shashwatj07) [reply src/test/java/com/puppycrawl/tools/checkstyle/api/AutomaticBeanTest.java]: done

**Review by pbludov — APPROVED** [review]

**Pull request merged**